lib/core/text/icu/caseModification.nit

```diff
@@ -0,0 +1,181 @@
+# This file is part of NIT ( http://www.nitlanguage.org ).
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Adding locale aware case modification to the String and CString classes
+
+module caseModification is pkgconfig("icu-io", "icu-i18n", "icu-uc")
+import locale
+
+`{
+	#include <unicode/utypes.h>
+	#include <unicode/ustring.h>
+
+	#include <string.h>
+	#include <locale.h>
+
+`}
+
+
+
+# There are 3 different types of case modification but they all behave the same way
+# Therefore the main operations are in a parent class
+private abstract class CaseModifier
+
+	fun getModifiedStrLength (uchar_str, error : Pointer, strLength : Int, localeName : CString) : Int  is abstract
+	fun doModification  (uchar_modifiedStr, uchar_str, error : Pointer, requiredLength, strLength : Int, localeName : CString) is abstract
+
+	fun modify (locale : Locale, str : CString) : CString import String.to_cstring, CString.cstring_length, Locale.getName, getModifiedStrLength, doModification `{
+
+
+		UErrorCode error = U_ZERO_ERROR;
+		char * localeName = NULL;
+		char * modifiedStr = NULL;
+		UChar * uchar_str = NULL;
+		UChar * uchar_modifiedStr = NULL;
+		int32_t requiredLength, strLength;
+
+		strLength = CString_cstring_length(str);
```

Comment:
I don't know what is ICU's behaviour regarding `\0`s in a string, but as they are valid in a Nit string, I'd refrain from using `strlen` as much as possible as we risk having a truncated string as result of a call to ICU

Reply:
As much as possible, I'd defer most of the bookkeeping (string length, references, etc.) to Nit, as most of the time, these CString functions will be called from a container class (Text and its derivatives), that already know the byte-size of a string.
This can be leveraged to pass the information to the lower levels, simplifying the C code, and removing much of the imports of the functions you define here.